resolver/serviceResolver.go

```diff
@@ -269,6 +269,82 @@ func (resolver *serviceResolver) SignMessage(userIp string, request requests.Sig
 
 }
 
+// SetSecurityModeNoExpire gets the user's guardian, verifies the codes and then sets the SecurityMode
+func (resolver *serviceResolver) SetSecurityModeNoExpire(userIp string, request requests.SetSecurityModeNoExpireMessage) (*requests.OTPCodeVerifyData, error) {
+	userAddress, err := sdkData.NewAddressFromBech32String(request.UserAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	addressBytes := userAddress.AddressBytes()
+	resolver.userCritSection.Lock(string(addressBytes))
+	defer resolver.userCritSection.Unlock(string(addressBytes))
+
+	userInfo, err := resolver.getUserInfo(addressBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	ctxGetGuardianData, cancelGetGuardianData := context.WithTimeout(context.Background(), resolver.requestTime)
+	defer cancelGetGuardianData()
+	guardianData, err := resolver.httpClientWrapper.GetGuardianData(ctxGetGuardianData, request.UserAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	guardianAddrBytes, err := resolver.pubKeyConverter.Decode(guardianData.ActiveGuardian.Address)
```

Comment:
before this line, we should do one more check. The account may not have an active guardian, thus returning a nil pointer, which will lead to a panic on this line when trying to access it:

```go
	if !check.IfNilReflect(guardianData.ActiveGuardian) {
		return nil, ErrAccountHasNoActiveGuardian
	}
```
the new error should be added in resolver/errors.go

Reply:
I had this problem when there wasn't a guardian for my account, thought about adding a check but totally forgot :)
updated